### ISSUE TITLE
Extension assembly isolation

### ISSUE THREAD

**Issue by veler** [DevToys-app/DevToys#1430]
### Current behavior

DevToys relies on some dependencies, such as `NuGet.Versioning`, version `6.9`.

If I create a DevToys extension that relies on a different version of `NuGet.Versioning`, such as `5.0` or `6.11`, it is ignored, and instead, the version `6.9` is used.

This causes a major problem when an extension tries to invoke a feature from `6.11` that does not exist in `6.9`.

See discussion here: https://github.com/DevToys-app/DevToys/discussions/1333#discussioncomment-11053480

### How to reproduce it (as minimally and precisely as possible)

Create a DevToys extension that uses `Nuget.Versioning` version `6.11`.

### Expected behavior

it should uses the NuGet version I used in my extension

### Screenshots

![image](https://github.com/user-attachments/assets/44e1240a-a754-43d4-b322-52163d7016ed)

![image](https://github.com/user-attachments/assets/a7c008b1-8857-4e08-80ad-c3eeada0505d)

### Workaround

Use the same version than DevToys.

### Affected DevToys kind

DevToys (app with GUI)

### DevToys Version

2.0.5.0

**Comment by jerone** [comment]
Sadly the fix was reverted in #1449 as it was not compatible for all platforms.

Can we reopen this issue and is there a potential different fix?

Is this even an issue for Mac?

**Comment by veler** [comment]
Indeed, sorry about that. It was causing a regression on Mac so I quickly reverted the change. And yes, this issue also applies on Mac and Linux.
